genson/src/main/java/com/owlike/genson/convert/CircularClassReferenceConverterFactory.java

```diff
@@ -51,10 +61,14 @@ void setDelegateConverter(Converter<T> delegate) {
     } else {
       try {
         CircularConverter circularConverter = new CircularConverter();
-        _circularConverters.get().put(type, circularConverter);
-        Converter converter = next().create(type, genson);
-        circularConverter.setDelegateConverter(converter);
-        return converter;
+        try {
+          _circularConverters.get().put(type, circularConverter);
+          Converter converter = next().create(type, genson);
+          circularConverter.setDelegateConverter(converter);
+          return converter;
+        } finally {
+          circularConverter.initLatch.countDown();
```

Comment:
I think that a simpler alternative would be to just make the create method synchronized (as locks are reentrant).

Or to get fancier it could be a lock on the type. However considering that this creation code is not invoked often but mostly the first time this type needs to be ser/de, a simple synchronized on the method is enough IMO.

This bug makes me wonder if having a ThreadSafeConverterFactory wrapper wouldn't be a good idea... we could systematically wrap the root factory in it to make sure we don't hit other initialization race bugs. Thoughts?